### PR TITLE
feat(internal/provider): add env_map to cached_image_resource

### PR DESCRIPTION
**Pull request by johnstcn**
This PR adds `env_map` to `cached_image_resource`. This consists of the computed env in map format, which can be useful for other providers that do not expect `KEY=VALUE` format.

### CODE REVIEW

File: internal/provider/cached_image_resource.go [hunk 1]
```diff
@@ -226,16 +230,24 @@ func (r *CachedImageResource) Schema(ctx context.Context, req resource.SchemaReq
 			},
 
 			// Computed "outputs".
-			// TODO(mafredri): Map vs List? Support both?
 			"env": schema.ListAttribute{
-				MarkdownDescription: "Computed envbuilder configuration to be set for the container. May contain secrets.",
+				MarkdownDescription: "Computed envbuilder configuration to be set for the container in the form of a list of stringss of `key=value`. May contain secrets.",
```
Comment (johnstcn) [inline internal/provider/cached_image_resource.go:234]: :snake: :snake: :snake:
```suggestion
				MarkdownDescription: "Computed envbuilder configuration to be set for the container in the form of a list of strings of `key=value`. May contain secrets.",
```

File: internal/provider/cached_image_resource.go [hunk 2]
```diff
@@ -396,29 +416,36 @@ func (r *CachedImageResource) Create(ctx context.Context, req resource.CreateReq
...
+		env["ENVBUILDER_GIT_PASSWORD"] = tfValueToString(data.GitPassword)
 	}
 	// Default to remote build mode.
 	if data.RemoteRepoBuildMode.IsNull() {
-		data.Env = appendKnownEnvToList(data.Env, "ENVBUILDER_REMOTE_REPO_BUILD_MODE", types.BoolValue(true))
+		env["ENVBUILDER_REMOTE_REPO_BUILD_MODE"] = "true"
 	} else {
-		data.Env = appendKnownEnvToList(data.Env, "ENVBUILDER_REMOTE_REPO_BUILD_MODE", data.RemoteRepoBuildMode)
+		env["ENVBUILDER_REMOTE_REPO_BUILD_MODE"] = tfValueToString(data.RemoteRepoBuildMode)
 	}
```
Comment (mafredri) [inline internal/provider/cached_image_resource.go:441]: Side-note: I feel like we're missing some input parameters here 🤔

**Review by mafredri — APPROVED** [review]
Nice!

File: internal/provider/cached_image_resource.go (hunk 2 above)
Comment (johnstcn) [reply internal/provider/cached_image_resource.go:441]: Yeah I need to go through and add all the rest! Follow up though.

**Pull request merged**